flowmachine/flowmachine/core/server/query_proxy.py

```diff
@@ -139,7 +135,70 @@ def construct_query_object(query_kind, params):  # pragma: no cover
             )
         except Exception as e:
             raise QueryProxyError(f"{error_msg_prefix}: '{e}'")
+    elif "location_event_counts" == query_kind:
+        start_date = params["start_date"]
+        end_date = params["end_date"]
+        interval = params["interval"]
+        level = params["aggregation_unit"]
+        subscriber_subset = params["subscriber_subset"]
+        direction = params["direction"]
+        event_types = params["event_types"]
+
+        error_msg_prefix = f"Error when constructing query of kind {query_kind} with parameters {params}"
+        allowed_intervals = TotalLocationEvents.allowed_levels
```

Comment:
A bit confusing that this is called `allowed_levels` and not `allowed_intervals`, but not a problem.